rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -112,10 +128,19 @@ bool SequentialReader::has_next()
   throw std::runtime_error("Bag is not open. Call open() before reading.");
 }
 
+void SequentialReader::decompress_message(
+  rosbag2_storage::SerializedBagMessage * message)
+{
+  if (compression_mode_ == CompressionMode::MESSAGE) {
+    decompressor_->decompress_serialized_bag_message(message);
+  }
+}
+
 std::shared_ptr<rosbag2_storage::SerializedBagMessage> SequentialReader::read_next()
 {
   if (storage_) {
     auto message = storage_->read_next();
```

Comment:
Should message be const?

Reply:
No, we decompress it in place. It shouldn't be const